include/kernel.h

```diff
@@ -787,9 +787,11 @@ void k_thread_system_pool_assign(struct k_thread *thread);
  *
  * @param duration Number of milliseconds to sleep.
  *
- * @return N/A
+ * @return Zero if the requested time has elapsed or the number of miliseconds
+ * left to sleep, if thread was woken up by \ref k_wakeup call.
+ *
  */
-__syscall void k_sleep(s32_t duration);
+__syscall s32_t k_sleep(s32_t duration);
```

Comment:
Why return value is signed here? Is it possible that thread actually slept for -100 milliseconds?

Reply:
I assume to return -Exxx

Reply:
I just followed all other zephyr code which uses s32_t for time.